ERCS/erc-7629.md

```diff
@@ -0,0 +1,296 @@
+    event ERC20ToERC721(address indexed to, uint256 amount, uint256[] tokenIds);
+
+    // ERC-721 to ERC-20 Conversion event
+    event ERC721ToERC20(address indexed to, uint256 tokenId, uint256 amount);
+
+    /**
+     * @dev Returns the name of the token.
+     */
+    function name() external view returns (string memory);
+
+    /**
+     * @dev Returns the symbol of the token.
+     */
+    function symbol() external view returns (string memory);
+
+    /**
+     * @dev Returns the number of decimals used in the token.
+     */
+    function decimals() external view returns (uint8);
+
+    /**
+     * @dev Returns the total supply of the ERC-20 tokens.
+     */
+    function totalSupply() external view returns (uint256);
+
+    /**
+     * @dev Returns the balance of an address for ERC-20 tokens.
+     * @param owner The address to query the balance of.
+     */
+    function balanceOf(address owner) external view returns (uint256);
+
+    /**
+     * @dev Returns the total supply of ERC-20 tokens.
+     */
+    function erc20TotalSupply() external view returns (uint256);
+
+    /**
+     * @dev Returns the balance of an address for ERC-20 tokens.
+     * @param owner The address to query the balance of.
+     */
+    function erc20BalanceOf(address owner) external view returns (uint256);
+
+    /**
+     * @dev Returns the total supply of ERC-721 tokens.
+     */
+    function erc721TotalSupply() external view returns (uint256);
+
+    /**
+     * @dev Returns the balance of an address for ERC-721 tokens.
+     * @param owner The address to query the balance of.
+     */
+    function erc721BalanceOf(address owner) external view returns (uint256);
+
+    /**
+     * @dev Checks if an operator is approved for all tokens of a given owner.
+     * @param owner The address of the token owner.
+     * @param operator The address of the operator to check.
+     */
+    function isApprovedForAll(
+        address owner,
+        address operator
+    ) external view returns (bool);
+
+    /**
+     * @dev Returns the remaining number of tokens that spender will be allowed to spend on behalf of owner.
+     * @param owner The address of the token owner.
+     * @param spender The address of the spender.
+     */
+    function allowance(
+        address owner,
+        address spender
+    ) external view returns (uint256);
+
+    /**
+     * @dev Returns the array of ERC-721 token IDs owned by a specific address.
+     * @param owner The address to query the tokens of.
+     */
+    function owned(address owner) external view returns (uint256[] memory);
+
+    /**
+     * @dev Returns the address that owns a specific ERC-721 token.
+     * @param id The token ID.
+     */
+    function ownerOf(uint256 id) external view returns (address erc721Owner);
+
+    /**
+     * @dev Returns the URI for a specific ERC-721 token.
+     * @param id The token ID.
+     */
+    function tokenURI(uint256 id) external view returns (string memory);
```

Comment:
```suggestion
    function tokenURI(uint256 tokenId) external view returns (string memory);
```
To avoid ambitious, all ERC721 related `id` should be consistently labelled as `tokenId`